code/modules/mob/mob.dm

```diff
@@ -230,7 +233,24 @@
 			del(narglow)
 
 /mob/proc/see_rift(var/obj/machinery/singularity/narsie/large/exit/R)
-	if((R.z == src.z) && (get_dist(R,src) <= (R.consume_range+10)) && !(R in view(src)))
+	if(istype(src.loc,/obj/mecha/))
```

Comment:
There's a lot of copy-paste which seems to me largely centred around the fact that the mob isn't always on the turf. So why not just use get_turf and bypass it, instead of accounting for specific situations?